src/main/java/Bob.java

```diff
@@ -0,0 +1,221 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+
+public class Bob {
+    public static void main(String[] args) {
+        String greet = " Hello! I'm Bob.\n"
+                + " What can I do for you?\n";
+
+        String exit = " Bye. Hope to see you again soon!";
+
+        System.out.println(greet);
+
+        Scanner scanner = new Scanner(System.in);
+        ArrayList<Task> taskList = new ArrayList<>(100);
+
+        while (true) {
+            String input = scanner.nextLine();
+
+            if (input.equals("bye")) {
+                System.out.println(exit);
+                break;
+            }
+
+            else if (input.equals("list")) {
+                int size = taskList.size();
+
+                System.out.println(" Here are the tasks in your list:");
+
+                for (int count = 0; count < size; count++) {
+                    System.out.println(" " + (count + 1) + "." + taskList.get(count));
+                }
+            }
+
+            else if (input.trim().matches("mark|unmark|deadline|todo|event|delete")) {
+                System.out.println(" Your entry is incomplete!");
+            }
+
+            else if (input.startsWith("mark ")) {
+                try {
+                    int index = Integer.parseInt(input.substring(5)) - 1;
+                    Task task = taskList.get(index);
+                    task.markAsDone();
+
+                    System.out.println(" Nice! I've marked this task as done:\n"
+                            + "  " + task);
+                }
+
+                catch (IndexOutOfBoundsException e) {
+                    System.out.println(" There exists no such task.");
+                }
+            }
+
+            else if (input.startsWith("unmark ")) {
+                try {
+                    int index = Integer.parseInt(input.substring(7)) - 1;
+                    Task task = taskList.get(index);
+                    task.markAsUndone();
+
+                    System.out.println(" OK, I've marked this task as not done yet:\n"
+                            + "  " + task);
+                }
+
+                catch (IndexOutOfBoundsException e) {
+                    System.out.println(" There exists no such task.");
+                }
+            }
+
+            else if (input.startsWith("deadline ")) {
+                int byIndex = input.indexOf("/by ");
+                String taskDescription = input.substring(9, byIndex).trim();
+                String deadlineDate = input.substring(byIndex + 4).trim();
+
+                Task newTask = new Deadline(taskDescription, deadlineDate);
+                taskList.add(newTask);
+                System.out.println(" Got it. I've added this task:\n"
+                        + "  " + newTask + "\n"
+                        + " Now you have " + taskList.size() + " tasks in the list.");
+            }
+
+            else if (input.startsWith("todo ")) {
```

Comment:
Having spaces at the end of the string does not look very clean. Maybe can consider removing that.